include/xtensor/xoperation.hpp

```diff
@@ -959,6 +959,74 @@ namespace xt
     {
         return detail::make_xfunction<typename detail::cast<R>::functor>(std::forward<E>(e));
     }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class T>
+    inline auto isin(E&& e, std::initializer_list<T> values) noexcept
+    {
+        auto lambda = [values](const auto& t) -> decltype(e == e) {
+            return std::find(values.begin(), values.end(), t) != values.end(); };
+        return make_lambda_xfunction(std::move(lambda), std::forward<E>(e));
+    }
+
+    /**
+    * @ingroup logical_operators
+    * @brief isin
+    *
+    * Returns a boolean array of the same shape as element that is ``true`` where an element of
+    * ``element`` is in ``test_elements`` and ``False`` otherwise.
+    * @param e an \ref xexpression
+    * @return a boolean array
+    */
+    template <class E, class F>
+    inline auto isin(E&& e, F&& values) noexcept
+    {
+        auto lambda = [values](const auto& t) -> decltype(e == e) {
```

Comment:
You may want to optimize the capture here, to avoid copies when `values` is a lvalue-reference. A solution for that is to introduce an intermediate structure specialized on `F` lvalueness and returning the lambda with capture by value or by reference.

Reply:
Finally I don't know how to do this. I guess there should be a `struct` with two `operator(const auto& t)` overloads, one is selected of `lvalue`s. However, where does the capture enter in this?